components/brave_wallet/browser/blockchain_registry.cc

```diff
@@ -116,35 +117,68 @@ void BlockchainRegistry::GetAllTokens(const std::string& chain_id,
   std::move(callback).Run(std::move(tokens_copy));
 }
 
-void BlockchainRegistry::GetBuyTokens(mojom::OnRampProvider provider,
-                                      const std::string& chain_id,
-                                      GetBuyTokensCallback callback) {
+std::vector<brave_wallet::mojom::BlockchainTokenPtr>
+BlockchainRegistry::GetBuyTokens(
```

Comment:
Thanks @yrliou, handling it this way using a `flat_set` will help a lot. What caused me trouble were duplications related to contract addresses declared all uppercase for some providers, and all lowercase for others.
For example, check the Basic Attention Token on mainnet.
For the OnRamp.Network the contract address is declared all lowercase: https://github.com/brave/brave-core/blob/c73d16cc1f77ec351edd402552549f1d93980109/components/brave_wallet/browser/brave_wallet_constants.cc#L158-L160

While for Sardine it's all uppercase:
https://github.com/brave/brave-core/blob/c73d16cc1f77ec351edd402552549f1d93980109/components/brave_wallet/browser/brave_wallet_constants.cc#L359-L361

This of course led to duplication in the UI, and I've eventually implemented a method in the Java layer to remove duplicated entries. Not sure if there's an easy way to handle this at C++ level, if you have suggestions I'll be happy to include them. Otherwise, we can leave it as is and keep the duplication check in Java.

Reply:
I don't think the duplicate tokens needs to be blocking this PR as other frontends are handling duplicates already too, we can open a follow-up issue to return a map (since mojo doesn't support set afaik) instead of array to help mitigate this all in backend.

Reply:
Fine by me 👍